WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2656,6 +2656,13 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                         mOnGetMentionResult = null;
                     }
                     break;
+                case RequestCodes.DOCUMENT_LIBRARY:
+                    List<Uri> documentUris = WPMediaUtils.retrieveMediaUris(data);
```

Comment:
This might be a little confusing but we were following the previous implementation so we didn't want to break the previous behaviour. Your implementation (with `retrieveMediaUris`) works for Android system picker. 
You should have another case to read the data from the media picker like this: 
```
if (data.hasExtra(MediaPickerConstants.EXTRA_MEDIA_URIS)) {
                        List<Uri> uris = convertStringArrayIntoUrisList(
                                data.getStringArrayExtra(MediaPickerConstants.EXTRA_MEDIA_URIS));
}
```
Basically if your request code is `PHOTO_PICKER` (with the media picker flag turned on), it should just work

Reply:
Thanks for sharing these thoughts @planarvoid I implemented this behavior here https://github.com/wordpress-mobile/WordPress-Android/pull/13413